### ISSUE TITLE
Speech-to-text Listening issue

### ISSUE THREAD

**Issue by Saraswathi24** [IBM/watson-voice-bot#62]
![image](https://user-images.githubusercontent.com/53259433/93445331-9a214d00-f8ee-11ea-9495-c2e7f05b9d8e.png)
 I have followed all the steps mentioned  in Readme.nd and tried to deploy the code locally. I haven't installed any web socket yet.
With the default one, I am getting output as follows:
INFO:engineio.server:Server initialized for threading.
INFO:assistant_setup:Using WORKSPACE_ID=cd761ded-c1a9-476a-89c4-c6de53e00ce2
WARNING:werkzeug:WebSocket transport not available. Install eventlet or gevent and gevent-websocket for improved performance.
 * Serving Flask app "app" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
INFO:werkzeug: * Running on http://192.168.29.211:5000/ (Press CTRL+C to quit)
INFO:werkzeug:192.168.29.211 - - [17/Sep/2020 14:02:58] "GET /api/conversation HTTP/1.1" 200 -
INFO:werkzeug:192.168.29.211 - - [17/Sep/2020 14:03:05] "POST /api/text-to-speech HTTP/1.1" 200 -

The program isn't working when the mic button is clicked it is just enabling the listening mode, after that webpage isn't responding anything. Kindly help me with this issue.

**Comment by markstur** [comment]
@sanjeevghimire Can you help with this?

**Comment by sanjeevghimire** [comment]
@Saraswathi24 the code is working mostly in firefox and chrome. I tested it on mine and the mic works. Can you make sure the version chrome and firefox you use has support for `navigator.getUserMedia` and is not deprecated?

**Comment by Saraswathi24** [comment]
@sanjeevghimire  The  `navigator.getUserMedia()` is not deprecated in Chrome, even then I find difficulty in recording the audio.It is still not working as expected.

**Comment by SapnaChouta** [comment]
@sanjeevghimire - We have similar issues.  Voice bot shows as "listening" when we talk but does not respond back.. We have tried this from multiple Mac laptops and multiple browsers like Firefox and Chrome. navigator.getUserMedia() is not deprecated.

**Comment by edffrench** [comment]
I'm having the same issue running off a ThinkPad, I've tried chrome and firefox. I've attached the respective DevTools consoles for the browsers.

![chrome console](https://user-images.githubusercontent.com/71700396/103875408-49ff0e00-50ca-11eb-83bf-2897e0dde3b4.png)
![chrome console](https://user-images.githubusercontent.com/71700396/103875416-4c616800-50ca-11eb-9d65-4a0ad2e54ee0.png)

**Comment by yeshapatel356** [comment]
Having same issue.

**Comment by yeshapatel356** [comment]
Having same issue.

**Comment by yeshapatel356** [comment]
Having same issue.

**Comment by erum007** [comment]
It worked on my PC but when I did the exact same thing from scratch on laptop, I face this issue and cannot figure it out.

**Comment by erum007** [comment]
OK so I found a solution for those who still need it. First of all, use this repository: https://github.com/yurilq/voice_bot
Also, after you configure it on one device, do not use the same credentials on another device.

**Comment by markstur** [comment]
I edited the above comment from @erum007 suggesting to use another repository because I don't see any updates there.  It is just a copy of this one.  Please clarify if there is an actual fix.  Maybe there is a point-in-time or commit in this repo you are referring to?  Please don't just redirect folks to another repository.

**Comment by markstur** [comment]
The comment about not using multiple devices is a good tip though.  I think this bot gets confused easily and having multiple connections from wherever probablky doesn't help.

**Comment by erum007** [comment]
> I edited the above comment from @erum007 suggesting to use another repository because I don't see any updates there. It is just a copy of this one. Please clarify if there is an actual fix. Maybe there is a point-in-time or commit in this repo you are referring to? Please don't just redirect folks to another repository.

I don’t know man, I was just playing around. Found that one and it worked while this didn’t. That one is outdated while this one is relatively updated so there must be a problem with any update made. I didn’t dig deep as to what specifically was the problem.

**Comment by markstur** [comment]
Right, and thanks for the info.  I think this issue is mostly about browser compatibility, but people are also getting caught in some other problems (e.g. multiple simultaneous connections?) that have not been well reproduced.  With the latest version I've seen some "works for me" responses.  I'll put the commit which was used in that "other repo" in this comment.  It's from 2018, but if anyone wants to compare an old version w/ the latest I don't want to lose that info. The latest commit used above was:

https://github.com/IBM/watson-voice-bot/commit/e3b2f1db25a13056d2c7c43bc120823af9ee96f6

**Comment by ShuwaiGhz** [comment]
Has anyone fix this issue? I am currently having the same issue. Hope anyone can give me pointer on this. It works fine at first, but now it cannot listen what I speak. I tried using firefox , ie, edge as well. but nothing works
